src/main/java/aeneas/views/BullpenView.java

```diff
@@ -4,37 +4,42 @@
 
 import com.jfoenix.effects.JFXDepthManager;
 
+import aeneas.models.Bullpen;
 import aeneas.models.Model;
 import aeneas.models.Piece;
 
 import javafx.geometry.Pos;
 import javafx.scene.layout.Pane;
 import javafx.scene.layout.VBox;
 
-class BullpenView {
+public class BullpenView {
 
   VBox bullpenBox;
-  Pane levelView ;
+  Pane levelView;
 
-  private static final int SQUARE_SIZE = 16;
+  static final int SQUARE_SIZE = 16;
 
   ArrayList<Pane> values = new ArrayList<Pane>();
 
-  BullpenView(VBox bullpenBox, Pane levelView){
+  public BullpenView(VBox bullpenBox, Pane levelView){
     this.levelView = levelView;
     this.bullpenBox = bullpenBox;
     JFXDepthManager.setDepth(bullpenBox, 1);
     bullpenBox.setAlignment(Pos.TOP_CENTER);
   }
 
-  void addPiece(Piece piece, Model model){
-    Pane piecePane = new Pane();
-    PieceView pieceView = new PieceView(levelView, piece, model, SQUARE_SIZE);
-    pieceView.setId(piece.toString()); //this relies on all instances having different to strings
-    piecePane.getChildren().add(pieceView);
-    values.add(piecePane);
-    bullpenBox.getChildren().add(piecePane);
-  }
+  public void refresh(Model model, Bullpen bullpen) {
 
+    bullpenBox.getChildren().clear();
 
+    for (int i=bullpen.getPieces().size() - 1; i >= 0; i-- ) {
+      Piece piece = bullpen.getPieces().get(i);
+      Pane piecePane = new Pane();
+      PieceView pieceView = new PieceView(levelView, piece, model, SQUARE_SIZE);
+      pieceView.setId(piece.toString()); //this relies on all instances having different to strings
```

Comment:
No that's an artifact--it is not used. Something like that will be needed on-drop, but it will be using a difference method. I will delete this